charts/consul/test/acceptance/tests/partitions/partitions_test.go

```diff
@@ -0,0 +1,148 @@
+
+	ctx := context.Background()
+
+	primaryHelmValues := map[string]string{
+		"global.datacenter": "dc1",
+		"global.image":      "hashicorp/consul-enterprise:1.11.0-ent-alpha",
+
+		"global.adminPartitions.enabled": "true",
+		"global.enableConsulNamespaces":  "true",
+		"global.tls.enabled":             "true",
+
+		"server.exposeGossipAndRPCPorts": "true",
+
+		"connectInject.enabled": "true",
+		"controller.enabled":    "true",
+	}
+
+	if cfg.UseKind {
+		primaryHelmValues["global.adminPartitions.service.type"] = "NodePort"
+		primaryHelmValues["global.adminPartitions.service.nodePort.https"] = "30000"
+	}
+
+	releaseName := helpers.RandomName()
+
+	// Install the primary consul cluster in the default kubernetes context
+	primaryConsulCluster := consul.NewHelmCluster(t, primaryHelmValues, primaryContext, cfg, releaseName)
+	primaryConsulCluster.Create(t)
+
+	// Get the TLS CA certificate and key secret from the primary cluster and apply it to secondary cluster
+	tlsCert := fmt.Sprintf("%s-consul-ca-cert", releaseName)
+	logger.Logf(t, "retrieving ca cert secret %s from the primary cluster and applying to the secondary", tlsCert)
+	caCertSecret, err := primaryContext.KubernetesClient(t).CoreV1().Secrets(primaryContext.KubectlOptions(t).Namespace).Get(ctx, tlsCert, metav1.GetOptions{})
+	caCertSecret.ResourceVersion = ""
+	require.NoError(t, err)
+	_, err = secondaryContext.KubernetesClient(t).CoreV1().Secrets(secondaryContext.KubectlOptions(t).Namespace).Create(ctx, caCertSecret, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	tlsKey := fmt.Sprintf("%s-consul-ca-key", releaseName)
+	logger.Logf(t, "retrieving ca key secret %s from the primary cluster and applying to the secondary", tlsKey)
+	caKeySecret, err := primaryContext.KubernetesClient(t).CoreV1().Secrets(primaryContext.KubectlOptions(t).Namespace).Get(ctx, tlsKey, metav1.GetOptions{})
+	caKeySecret.ResourceVersion = ""
+	require.NoError(t, err)
+	_, err = secondaryContext.KubernetesClient(t).CoreV1().Secrets(secondaryContext.KubectlOptions(t).Namespace).Create(ctx, caKeySecret, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	var partitionSvcIP string
+	if !cfg.UseKind {
+		// Get the IP of the partition service to configure the external server address in the values file for the workload cluster.
+		partitionServiceName := fmt.Sprintf("%s-partition-secret", releaseName)
+		logger.Logf(t, "retrieving partition service to determine external IP for servers")
+		partitionsSvc, err := primaryContext.KubernetesClient(t).CoreV1().Services(primaryContext.KubectlOptions(t).Namespace).Get(ctx, partitionServiceName, metav1.GetOptions{})
+		require.NoError(t, err)
+		partitionSvcIP = partitionsSvc.Status.LoadBalancer.Ingress[0].IP
+	} else {
+		nodeList, err := primaryContext.KubernetesClient(t).CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		require.NoError(t, err)
+		for _, node := range nodeList.Items {
+			partitionSvcIP = node.Status.Addresses[0].Address
```

Comment:
since it's kind should we assume that there's only one node and skip the for loop? I think the for loop reads a bit confusing.

Reply:
That makes sense. I will remove the loop and just grab the 0th address and add a comment.